turbo/snapshotsync/freezeblocks/block_reader.go

```diff
@@ -373,34 +375,61 @@ func (r *RemoteBlockReader) CanonicalBodyForStorage(ctx context.Context, tx kv.G
 	}
 	return body, nil
 }
-func (r *RemoteBlockReader) Checkpoint(ctx context.Context, tx kv.Getter, spanId uint64) ([]byte, error) {
-	return nil, nil
+func (r *RemoteBlockReader) Checkpoint(ctx context.Context, tx kv.Tx, spanId uint64) (*heimdall.Checkpoint, bool, error) {
+	return nil, false, nil
 }
 
 // BlockReader can read blocks from db and snapshots
 type BlockReader struct {
-	sn    *RoSnapshots
-	borSn *BorRoSnapshots
+	sn             *RoSnapshots
+	borSn          *heimdall.RoSnapshots
+	borBridgeStore bridge.Store
```

Comment:
These are being added so that temporarily block_reader can retain its interface to the stage code.  Doing it like this allows us to test the functionality in the bor specific code does not break the current stage code - hence it proves that it running successfully.

Once we remove the stages all of the bor specific code - bridge snapshots etc can be removed from here.